subprojects/logging/src/main/java/org/gradle/util/SingleMessageLogger.java

```diff
@@ -230,40 +314,97 @@ public static void nagUserOfReplacedNamedParameter(String parameterName, String
     }
 
     /**
-     * Try to avoid using this nagging method. The other methods use a consistent wording for when things will be removed.
+     * Output format:
+     * <p>
+     * ${summary} has been deprecated. This is scheduled to be removed in Gradle X. ${advice}
      */
-    public static void nagUserWithDeprecatedIndirectUserCodeCause(String summary) {
-        if (isEnabled()) {
-            nagUserWithDeprecatedIndirectUserCodeCause(summary, null);
-        }
-    }
-
     public static void nagUserWithDeprecatedBuildInvocationFeature(String summary, String advice) {
         nagUserWithDeprecatedBuildInvocationFeature(summary, thisWillBeRemovedMessage(), advice);
     }
 
+
+    /**
+     * Output format:
+     * <p>
+     * ${summary} has been deprecated. ${removeDetails} ${advice}
+     */
     public static void nagUserWithDeprecatedBuildInvocationFeature(String summary, String removalDetails, String advice) {
         nagUserWith(String.format("%s has been deprecated.", summary), removalDetails, advice, null, DeprecatedFeatureUsage.Type.BUILD_INVOCATION);
     }
 
+    /**
+     * Output format:
+     * <p>
+     * ${summary} has been deprecated. This is scheduled to be removed in Gradle X. ${advice}
+     */
     public static void nagUserWithDeprecatedIndirectUserCodeCause(String summary, @Nullable String advice) {
         if (isEnabled()) {
             nagUserWithDeprecatedIndirectUserCodeCause(summary, advice, null);
         }
     }
 
+    /**
+     * Try to avoid using this nagging method. The other methods use a consistent wording for when things will be removed.
```

Comment:
How is the message different from the other `nagUserWithDeprecatedIndirectUserCodeCause`? Should both be avoided?

Reply:
TBH, I don't think we need this many methods... but given that I'm just refactoring code, I decide not to make any change.